tests/Faktory/Test.hs

```diff
@@ -67,3 +70,11 @@ haltWorker :: Async a -> IO a
 haltWorker a = do
   withProducer $ \producer -> void $ perform @Text mempty producer "HALT"
   wait a
+
+flushQueue :: IO ()
+flushQueue = do
+  settings <- envSettings
+  workerSettings <- envWorkerSettings
+  a <- async $ runWorker settings workerSettings $ \faktoryJob ->
+    when (jobArg faktoryJob == ("HALT" :: Text)) $ throw WorkerHalt
+  haltWorker a
```

Comment:
Is there a reason to flush by actually consuming and not `withProducer $ void . flush` like we do for normal test cases in `startWorker` (see L52/54)?

Reply:
I didn't know that was an option!